test/functional/page_objects/visual_builder_page.js

```diff
@@ -82,6 +83,64 @@ export function VisualBuilderPageProvider({ getService, getPageObjects }) {
       await testSubjects.click('markdownDataBtn');
     }
 
+    /**
+     *
+     * getting all markdown variables list which located on `table` section
+     *
+     * **Note**: if `table` not have variables, return `No variables available for the selected data metrics.`
+     * @returns {Promise<string[] | string>}
+     * @memberof VisualBuilderPage
+     */
+    async getMarkdownTableVariables() {
+      const noVariablesSelector = '.euiTitle.euiTitle--xxsmall.tvbMarkdownEditor__noVariables';
+      const variablesListSelector = 'div.tvbMarkdownEditor__variables table > tbody';
+
+      const noVariables = await find.descendantExistsByCssSelector(noVariablesSelector);
+
+      if (!noVariables) {
+        const text = await (await find.byCssSelector(noVariablesSelector)).getVisibleText();
+        log.debug(`markdown table variable is: ${text}`);
+        return text;
+      }
+      const variables = await find.allByCssSelector(`${variablesListSelector} td`);
+      const variablesText = Promise.all(variables.map(variable => variable.getVisibleText()));
+      log.debug(`markdown table variables is: ${variablesText}`);
+      return variablesText;
+    }
+
+    /**
+     * get all sub-tabs count for `time series`, `metric`, `top n`, `gauge`, `markdown` or `table` tab.
+     *
+     * @returns {Promise<number>}
+     * @memberof VisualBuilderPage
+     */
+    async getSubTabsCount() {
+      const rootSelector = await find.byCssSelector('.euiTabs.euiTabs--small');
+      const tabsSelector = 'button[role="tab"]';
+      const elements = await rootSelector.findAllByCssSelector(tabsSelector);
+
+      return elements.length;
+    }
```

Comment:
with `data-test-subj` in each of the button, you can simply use `[data-test-subj$="TsvbTypeBtn"]`:

`return await find.findAllByCssSelector ('[data-test-subj$="TsvbTypeBtn"]')`
I would define function as `getTSVBTabs` and return list of ElementWrapper objects